### DISCUSSION TITLE
How do Actions ToS apply to self-hosted?

### DISCUSSION THREAD

**Discussion by mhart** [community/community#26922] [Actions]
<p>In the Actions Terms of Service:</p>
<p><a href="https://docs.github.com/en/github/site-policy/github-additional-product-terms#5-actions-and-packages" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/github/site-policy/github-additional-product-terms#5-actions-and-packages</a></p>
<p>It talks about the various things you can’t use “Actions” for, but the wording seems geared around paid-for hosted Actions.</p>
<p>Do these limitations still apply if you’re running your own self-hosted runner?</p>
<p>For example, the point:</p>
<blockquote>
<p>Actions should not be used for… the provision of a stand-alone or integrated application or service offering Actions or <em>any elements of Actions</em> for commercial purposes;</p>
</blockquote>
<p>If this were to apply to self-hosted runners, then it seems to imply cloud providers wouldn’t be able to charge users for the resources used to provision their own self-hosted runners.</p>
<p>Can GitHub confirm that these Terms of Service only apply to hosted Actions, and not self-hosted runners?</p>
<p>Thanks,</p>
<p>Michael</p>

**Comment by NicoNBW** [comment]
<p>Thanks for your feedback.</p>
<aside class="quote no-group" data-username="mhart" data-post="1" data-topic="131383">
<div class="title">
<div class="quote-controls"></div>
 mhart:</div>
<blockquote>
<p>If this were to apply to self-hosted runners, then it seems to imply cloud providers wouldn’t be able to charge users for the resources used to provision their own self-hosted runners.</p>
</blockquote>
</aside>
<p>As mentioned in <a href="https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#differences-between-github-hosted-and-self-hosted-runners" rel="nofollow noopener">this document</a>: Self-hosted runners are free to use with GitHub Actions, but you are responsible for the cost of maintaining your runner machines. And there is no distinction between the hosted runners and self-hosted runners in the document your shared, so the Terms of Service should apply to the entire GitHub Actions. Sorry for causing your confusion.</p>

**Comment by mhart** [comment]
<p>Hi Nico,</p>
<p>Thanks for getting back on this.</p>
<p>It’s still unclear what it means when it says “the provision of… any elements of Actions for commercial purposes” when applying to self-hosted though.</p>
<p>Does this mean that consultants can’t charge for their services to help clients setup self-hosted runners? That would seem very extreme.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>With regard to the listed restrictions on uses of Actions in section 5 of our Additional Product Terms, those terms are not primarily aimed at <strong>self-hosted</strong> runners, and rather focus on the burdens those restricted uses would place on our servers. Please note that all other areas of our Terms of Service, Additional Product Terms, Acceptable Use Policy and other policies do apply.</p>
<p>Additionally, if you are seeking an official GitHub answer to any question, especially with regard to our Terms of Service, only someone with a Staff title and Octocat badge on their avatar is authorized to supply one.</p>

**Comment by mhart** [comment]
<p>Thanks <a class="mention" href="/lee-dohm">@lee-dohm</a> – I figured they weren’t intended to stop me from mining cryptocurrency on my own machines, but just wanted to double-check! :smile:</p>
